### ISSUE TITLE
Add the possibility to define a command to be executed after the release of a modifier key

### ISSUE THREAD

**Issue by okraits** [i3/i3#3794]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
Currently, there's the `--release` flag which makes i3 execute the specified command when the specified key is released:

`bindsym [--release] [<Group>+][<Modifiers>+]<keysym> command`

## Desired Behavior
There should be the possibility to specify an additional command which will be executed when a specified modifier key is released.

Usecase:
I use a [script](https://github.com/altdesktop/i3ipc-python/blob/master/examples/i3-cycle-focus.py) to cycle through the list of most recently focused windows (I'd like to have a behaviour similar to the alt-tab function of openbox).
With that script, I keep the Alt key pressed and hit the Tab key several times until the desired window is focused. Then I also release the Alt key. Unfortunately, it's not possible to bind a command to the release of the Alt key (in this case to trigger the update of the focus history after the desired window has been selected). So I'm using a timer to update the focus history after a certain amount of time after the Tab key has been released.

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.13 (2016-11-08) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.13 (2016-11-08) (pid 17317) abort…)
Loaded i3 config: /home/okraits/.config/i3/config (Last modified: Mo 16 Sep 2019 17:31:34 CEST, 10365 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
- Linux Distribution & Version: Debian Stretch
- Are you using a compositor (e.g., xcompmgr or compton): None
</pre>

**Comment by xzfc** [comment]
> Usecase:
> I use a script to cycle through the list of most recently focused windows (I'd like to have a behaviour similar to the alt-tab function of openbox).

Have you tried using https://github.com/sagb/alttab?

**Comment by okraits** [comment]
> > Usecase:
> > I use a script to cycle through the list of most recently focused windows (I'd like to have a behaviour similar to the alt-tab function of openbox).
> 
> Have you tried using https://github.com/sagb/alttab?

No, as I don't see any sense in using third party tools when i3 provides an IPC interface to interact with it via scripts. It just doesn't expose the event needed for this usecase.

**Comment by okraits** [comment]
> Have you tried using https://github.com/sagb/alttab?

@xzfc I tried out alttab. In general, it's nice but it lacks one feature I want to have:

You can only define wether you want to switch between windows on all workspaces or only between windows on the current workspace.

But I would like to be able to switch between windows on all **visible** workspaces. So that windows on a not focused workspace on a second monitor are also included.

**Comment by xi** [comment]
I agree this would be a helpful enhancement. I can think of different ways to approach this:

1. Add special handling for the release of modifier keys
2. Add an option `--release-always`
3. Add a new command pair (e.g. `keysym` / `keycode`) that allows to bind directly to low level key events for single keys (no modifiers allowed)

Option (1) has the benefit of being restricted to modifier keys which is believe is sufficient. Options (2) and (3) have the benefit of being fully backwards compatible.
